File: 0530_Minimum Absolute Difference in BST/MinimumAbsoluteDifferenceinBST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

public class Solution {
    private int min = Integer.MAX_VALUE;
    private TreeNode prev;
    
    public int getMinimumDifference(TreeNode root) {
        if(root == null) return min;
        getMinimumDifference(root.left);
        
        if(prev!=null){
            min = Math.min(min,root.val - prev.val);
        }
        
        prev = root;
        
        getMinimumDifference(root.right);
        return min;
    }
}

File: 0530_Minimum Absolute Difference in BST/MinimumAbsoluteDifferenceinBST.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int GetMinimumDifference(TreeNode root) {
        var min = int.MaxValue;
        if(root == null) return min;
        
        
        var stack = new Stack<TreeNode>();
        TreeNode prev = null;
        
        while(root!=null || stack.Count()>0){
            while(root!=null){
                stack.Push(root);
                root = root.left;
            }
            
            root = stack.Pop();
            if(prev!=null){
                min = Math.Min(root.val - prev.val,min);
            }
            
            prev = root;
            root = root.right;
        }
        
        return min;
    }
}